Automattic/wp-calypso
# Frictionless Email Subscriptions: Subscribe user to mailing list

Related to https://github.com/Automattic/martech/issues/3098

## Proposed Changes

* If new user is successfully created when visiting the flow, subscribe user to guides mailing list
* If new user cannot be created because they already exist, subscribe user to guides mailing list
* Change path of flow from `/start/email-subscription/subscribing-email` to `/start/email-subscription/subscribe`
* Sanitize `redirect_to` query param to ensure that we only redirect to Automattic owned hosts

![2024-06-25 23 15 06](https://github.com/Automattic/wp-calypso/assets/5414230/7af45764-7d74-49ba-bd5c-86fac690233c)

## Why are these changes being made?


When someone subscribes to an email campaign on a landing page like wordpress.com/learn, we'd like to manage outbound communication PCYsg-6j2-p2. Guides, however, requires a user_id and a WordPress account.

In the past, if the user wanted to subscribe on a landing page and wasn't already logged in, they'd be redirected to the /log-in page. If they didn't have an account, they'd be prompted to create one. Once completed, they'd be redirected to the original landing page.

To polish this experience, we're creating a custom email subscription "signup" flow. For the happy path, the user should be able to subscribe with one click, and the technical implementation should look something like p5uIfZ-f11-p2#comment-22793.

## Testing Instructions



* Check out this branch
* Set the `signup/email-subscription-flow` flag to true in `config/development.json`
* Start local dev
* Navigate to `/start/email-subscription/subscribing?user_email={YOUR_NEW_EMAIL}&redirect_to=wordpress.com%2Flearn&mailing_list=learn` Note that the query params are required.
* Verify that a loading screen is initially rendered
* Replace the email query param with the email of a brand new user
* Verify that the new user is subscribed to the guides mailing list by locating the `/pigeon` fetch request in the developer console
* Revisit `/start/email-subscription/subscribing?user_email={YOUR_NEW_EMAIL}&redirect_to=wordpress.com%2Flearn&mailing_list=learn` to emulate an email subscription for an existing user. We will subscribe them to guides automatically behind the scenes in a future PR. A `/pigeon` fetch request in the developer console

## Pre-merge Checklist
